Networking/Networking/Network/NetworkError.swift

```diff
@@ -41,7 +42,13 @@ extension NetworkError {
         case StatusCode.timeout:
             self = .timeout
         default:
-            self = .unacceptableStatusCode(statusCode: statusCode)
+            let response: String? = {
+                guard let responseData else {
+                    return nil
+                }
+                return String(data: responseData, encoding: .utf8)
+            }()
```

Comment:
nit: a more concise version:

```suggestion
            let response: String? = responseData.map { String(data: $0, encoding: .utf8) } ?? nil
```